hydropandas/io/water_connect.py

```diff
@@ -0,0 +1,388 @@
+            "keep_all_obs is False"
+        )
+
+    if location_gdf is None:
+        location_gdf = get_locations_gdf()
+
+    if extent is not None:
+        location_gdf = get_locations_within_extent(location_gdf, extent)
+
+    if location_gdf.empty:
+        msg = f"No water connect measurements found within extent {extent}"
+        logger.warning(msg)
+        return []
+
+    nr_obs_wells = len(location_gdf)
+    logger.info(f"downloading waterconnect data from {nr_obs_wells} observation points")
+
+    obs_list = []
+    for dh_no, meta_series in tqdm(
+        location_gdf.iterrows(), total=nr_obs_wells, desc="obs well"
+    ):
+        o = ObsClass.from_waterconnect(
+            dh_no,
+            meta_series,
+            tmin=tmin,
+            tmax=tmax,
+            only_metadata=only_metadata,
+            **kwargs,
+        )
+        if o.empty and not keep_all_obs:
+            continue
+
+        obs_list.append(o)
+
+    return obs_list
+
+
+def get_locations_gdf(keep_cols="all", update=False):
+    """get locations of water drillholes in Southern Australia from water connect.
+
+    Parameters
+    ----------
+    keep_cols : list or str, optional
+        if provided only these columns of the geodataframe are stored. If 'all' all the
+        columns are returned, by default 'all'.
+    update : bool, optional
+        if True new locations are downloaded and stored locally even if the locations
+        were already downloaded before. By default False
+
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        locations of water drillholes in Southern Australia
+    """
+
+    # set directories
+    fdir = Path(user_data_dir("waterconnect", "hydropandas"))
+    fdir.mkdir(parents=True, exist_ok=True)
+    fname_feather = fdir / "WATER_Drillholes_GDA2020.feather"
+
+    # read files from feather
+    if fname_feather.exists() and not update:
+        logger.debug(f"Reading Water Connect locations from {fname_feather}")
+        gdf_sel = gpd.read_feather(fname_feather)
+        return gdf_sel
+
+    msg = (
+        "Initializing Water Connect, this may take several minutes. This only has to "
+        "be done the first time you use Water Connect via hydropandas."
+    )
+    logger.info(msg)
+
+    # download data
+    r = requests.get(DH_zip_url, timeout=3600)
+    r.raise_for_status()
+
+    # extract files from zip and write to disk
+    zip_buffer = BytesIO(r.content)
+    with zipfile.ZipFile(zip_buffer, "r") as z:
+        target_files = [
+            name for name in z.namelist() if name.startswith("WATER_Drillholes_GDA2020")
+        ]
+        # Extract only those files
+        logger.debug(f"Writing Water Connect locations shape to {fdir}")
+        for file in target_files:
+            z.extract(file, fdir)
+
+    # read shapefiles to geodataframe
+    logger.debug(f"Reading Water Connect locations from {fdir}")
+    gdf_gda2020 = gpd.read_file(fdir / "WATER_Drillholes_GDA2020.shp")
```

Comment:
you could also add a bbox here to get a subset of the points? Not sure it makes sense so just raising it as a suggestion.

Reply:
Yes I could do that but then you have to download all data twice (which takes approximately 4 minutes) if you want to read multiple extents. I think this is the fastest option

Reply:
Yes I could do that but then you have to download all data twice (which takes approximately 4 minutes) if you want to read multiple extents. I think this is the fastest option.